ui/src/tempVars/components/MapTemplateBuilder.tsx

```diff
@@ -120,39 +113,8 @@ class MapTemplateBuilder extends PureComponent<TemplateBuilderProps, State> {
 
   private constructValuesFromString(templateValuesString: string) {
     const {notify} = this.props
-    const trimmed = _.trimEnd(templateValuesString, '\n')
-    const parsedTVS = Papa.parse(trimmed)
-    const templateValuesData = getDeep<string[][]>(parsedTVS, 'data', [[]])
 
-    if (templateValuesData.length === 0) {
-      return
-    }
-
-    let arrayOfKeys = []
-    let values = []
-    _.forEach(templateValuesData, arr => {
-      if (arr.length === 2 || (arr.length === 3 && arr[2] === '')) {
-        const key = trimAndRemoveQuotes(arr[0])
-        const value = trimAndRemoveQuotes(arr[1])
-
-        if (!arrayOfKeys.includes(key) && key !== '') {
-          values = [
-            ...values,
-            {
-              type: TemplateValueType.Map,
-              value,
-              key,
-              selected: false,
-              localSelected: false,
-            },
-          ]
-          arrayOfKeys = [...arrayOfKeys, key]
-        }
-      } else {
-        notify(notifyInvalidMapType())
-      }
-    })
-    return values
+    return csvToMap(templateValuesString, () => notify(notifyInvalidMapType()))
```

Comment:
_nit_ passing callbacks for error handling feels a bit strange, maybe use a `try .. catch` statement instead? I think that when a callback is passed, it's not clear what the function returns just by reading the function invocation.

Reply:
I think I can modify this to return an object of errors and values, since the component tries to just skip over invalid keys, but notify if it encounters at least one invalid key-value pair

Reply:
oh I see, that makes sense. it's not try/catch since the function can partially succeed. 👍 👍